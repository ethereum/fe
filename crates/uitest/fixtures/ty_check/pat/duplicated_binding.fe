pub enum E {
    V(i32, u32)
}

fn foo(f: E) {
    let E::V(x, x) = f

    match f {
        E::V(x, x) => ()
    }
}
