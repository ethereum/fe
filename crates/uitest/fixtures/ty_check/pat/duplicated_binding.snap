---
source: crates/uitest/tests/ty_check.rs
expression: diags
input_file: crates/uitest/fixtures/ty_check/pat/duplicated_binding.fe
---
error[8-0002]: duplicated bidning `x` in the same pattern
  ┌─ duplicated_binding.fe:6:17
  │
6 │     let E::V(x, x) = f
  │              -  ^ `x` is already defined in the same pattern
  │              │   
  │              `x` is defined here

error[8-0002]: duplicated bidning `x` in the same pattern
  ┌─ duplicated_binding.fe:9:17
  │
9 │         E::V(x, x) => ()
  │              -  ^ `x` is already defined in the same pattern
  │              │   
  │              `x` is defined here


